library/core/src/time.rs

```diff
@@ -1701,3 +1765,121 @@ impl Duration {
         )
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+pub mod duration_verify {
+    use crate::kani;
+    use super::*;
+
+    impl kani::Arbitrary for Duration {
+        fn any() -> Duration {
+            let secs = kani::any::<u64>();
+            let nanos = kani::any::<u32>();
+            kani::assume(nanos < NANOS_PER_SEC || secs.checked_add((nanos / NANOS_PER_SEC) as u64).is_some());
+            Duration::new(secs, nanos)
+        }
+    }
+
+    #[kani::proof_for_contract(Duration::new)]
```

Comment:
do you mind adding a second harness for `duration_new` that does not use contract, and ensures that there's a panic instead? you can use `#[kani::should_panic]`. Thanks!

Reply:
Just added the should_panic harness.